device/transport/http/src/http.ts

```diff
@@ -384,7 +384,7 @@ export class Http extends EventEmitter implements DeviceTransport {
           drainRequester.on('nextRequest', () => {
             const request = this._http.buildRequest('GET', path + endpoint.versionQueryString(), httpHeaders, config.host, config.x509, (err, body, res) => {
               if (!err) {
-                if (body) {
+                if (res.statusCode === 200) {
```

Comment:
>200 [](start = 39, length = 3)

nit: maybe add a comment suggesting that 200 indicates a message and 204 indicates no message?